be/src/storage/storage_engine.cpp

```diff
@@ -1140,6 +1153,94 @@ void StorageEngine::increase_update_compaction_thread(const int num_threads_per_
     }
 }
 
+void StorageEngine::remove_increment_map_by_table_id(int64_t table_id) {
+    std::unique_lock delete_lock(_auto_increment_delete_mutex);
+
+    if (_tableid_auto_increment_map.find(table_id) != _tableid_auto_increment_map.end()) {
+        delete _tableid_auto_increment_map[table_id];
+        _tableid_auto_increment_map.erase(_tableid_auto_increment_map.find(table_id));
+    }
+
+    if (_tabletid_auto_increment_mutex.find(table_id) != _tabletid_auto_increment_mutex.end()) {
+        delete _tabletid_auto_increment_mutex[table_id];
+        _tabletid_auto_increment_mutex.erase(_tabletid_auto_increment_mutex.find(table_id));
+    }
+}
+
+Status StorageEngine::get_next_increment_id_interval(int64_t tableid, size_t num_row, std::vector<int64_t>& ids) {
+    bool need_init = false;
+    std::pair<int64_t, int64_t>* interval;
+    std::mutex* mutex;
+
+    // used for delete.
+    std::shared_lock delete_lock(_auto_increment_delete_mutex);
+    // lock shared mode to check <tableid, <min,max>> exist or not.
+    _auto_increment_mutex.lock_shared();
+    auto iter = _tableid_auto_increment_map.find(tableid);
+    auto end_iter = _tableid_auto_increment_map.end();
+
+    if (UNLIKELY(iter == end_iter)) {
+        _auto_increment_mutex.unlock_shared();
+        // lock mode to add the new <tableid, <min,max>>
+        _auto_increment_mutex.lock();
+
+        if (_tableid_auto_increment_map.find(tableid) == _tableid_auto_increment_map.end()) {
+            _tableid_auto_increment_map.insert({tableid, new std::pair<int64_t, int64_t>({0, 0})});
+            _tabletid_auto_increment_mutex.insert({tableid, new std::mutex()});
+            need_init = true;
+        }
+
+        interval = _tableid_auto_increment_map.at(tableid);
+        mutex = _tabletid_auto_increment_mutex.at(tableid);
+        _auto_increment_mutex.unlock();
+    } else {
+        interval = _tableid_auto_increment_map.at(tableid);
+        mutex = _tabletid_auto_increment_mutex.at(tableid);
+        _auto_increment_mutex.unlock_shared();
+    }
```

Comment:
have you test the performance gains of this optimization? Maybe just keep it simple first.

Reply:
Have changed to a simpler implementation